diff --git a/librmx/source/rmxbase/memory/String.h b/librmx/source/rmxbase/memory/String.h
index 27d0e78c..c8a3b6d1 100644
--- a/librmx/source/rmxbase/memory/String.h
+++ b/librmx/source/rmxbase/memory/String.h
@@ -168,8 +168,8 @@ public:
 
 	CLASS& operator=(const CLASS& str)			{ copy(str); return (CLASS&)*this; }
 	CLASS& operator=(const CHAR* str)			{ copy(str); return (CLASS&)*this; }
-	CLASS& operator=(const StdString& str)		{ copy(str); return *this; }
-	CLASS& operator=(const StdStringView& str)	{ copy(str); return *this; }
+	CLASS& operator=(const StdString& str)		{ copy(str); return (CLASS&)*this; }
+	CLASS& operator=(const StdStringView& str)	{ copy(str); return (CLASS&)*this; }
 
 	CLASS& operator<<(const CLASS& str)		{ add(str);			return (CLASS&)*this; }
 	CLASS& operator<<(CHAR ch)				{ add(ch);			return (CLASS&)*this; }
