diff --git a/librmx/source/rmxbase/memory/String.h b/librmx/source/rmxbase/memory/String.h
index a74aba75..ccc35913 100644
--- a/librmx/source/rmxbase/memory/String.h
+++ b/librmx/source/rmxbase/memory/String.h
@@ -69,8 +69,8 @@ public:
 	inline CHAR* accessData()				{ return mData; }
 	inline CHAR getChar(size_t index)		{ return (index < mLength) ? mData[index] : 0; }
 	inline CHAR getChar(int index)			{ return (index >= 0 && index < (int)mLength) ? mData[index] : 0; }
-	inline uint32 getUnicode(size_t index)	{ return toUnicode(getChar(index)); }
-	inline uint32 getUnicode(int index)		{ return toUnicode(getChar(index)); }
+	inline uint32 getUnicode(size_t index)	{ return StringTraits<CHAR>::toUnicode(getChar(index)); }
+	inline uint32 getUnicode(int index)		{ return StringTraits<CHAR>::toUnicode(getChar(index)); }
 
 	inline int length() const			{ return (int)mLength; }
 	inline bool empty() const			{ return (mLength == 0); }
@@ -97,7 +97,6 @@ public:
 	void addHex(unsigned int value);
 	void addFloat(float value, int precision = 0);
 	void addDouble(double value, int precision = 0);
-	void addDouble(double value);
 	void addData(void* inputdata, int bytes);
 
 	int parseInt() const;
@@ -169,8 +168,8 @@ public:
 
 	CLASS& operator=(const CLASS& str)			{ copy(str); return (CLASS&)*this; }
 	CLASS& operator=(const CHAR* str)			{ copy(str); return (CLASS&)*this; }
-	CLASS& operator=(const StdString& str)		{ copy(str); return *this; }
-	CLASS& operator=(const StdStringView& str)	{ copy(str); return *this; }
+	CLASS& operator=(const StdString& str)		{ copy(str); return (CLASS&)*this; }
+	CLASS& operator=(const StdStringView& str)	{ copy(str); return (CLASS&)*this; }
 
 	CLASS& operator<<(const CLASS& str)		{ add(str);			return (CLASS&)*this; }
 	CLASS& operator<<(CHAR ch)				{ add(ch);			return (CLASS&)*this; }
